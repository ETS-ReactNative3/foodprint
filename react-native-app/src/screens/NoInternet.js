import React from 'react';
import { View, Text, Image } from 'react-native';
import { StyleSheet } from 'react-native';

const NoInternet = () => {
  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        <Text h3 style={styles.title}>No Internet connection...</Text>
      </View>
      <View style={styles.body}>
        <Image
          style={styles.image}
          source={{ uri: 'https://mir-s3-cdn-cf.behance.net/project_modules/disp/d8094059120725.5a15c9fa09761.gif' }}
        />
      </View>
      <View style={styles.messageContainer}>
        <Text style={styles.message}>Turn off flight mode, move to a different location, or
            change mobile phone provider...</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    margin: 40,
  },
  titleContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  title: {
    fontSize: 36,
    textAlign: 'center',
    fontWeight: 'bold',
  },
  body: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  image: {
    height: 300,
    width: 300,
  },
  messageContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  message: {
    fontSize: 22,
    textAlign: 'center',
    margin: 20,
  },
});

export default NoInternet;
