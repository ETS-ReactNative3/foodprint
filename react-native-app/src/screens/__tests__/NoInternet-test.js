import React from 'react';
import NoInternet from '../NoInternet';

import renderer from 'react-test-renderer';

test('NoInternet matches previous snapshot', () => {
  const tree = renderer.create(<NoInternet />).toJSON();
  expect(tree).toMatchSnapshot();
});

