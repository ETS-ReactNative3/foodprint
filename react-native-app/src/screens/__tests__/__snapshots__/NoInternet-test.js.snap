// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoInternet matches previous snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "margin": 40,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      h3={true}
      style={
        Object {
          "fontSize": 36,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Oops.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://mir-s3-cdn-cf.behance.net/project_modules/disp/d8094059120725.5a15c9fa09761.gif",
        }
      }
      style={
        Object {
          "height": 300,
          "width": 300,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "margin": 20,
          "textAlign": "center",
        }
      }
    >
      You're not connected to the internet.
    </Text>
  </View>
</View>
`;
